core/src/main/java/io/micronaut/core/type/Argument.java

```diff
@@ -493,6 +493,19 @@ static <T> Argument<T> of(
         return new DefaultArgument<>(type, null, AnnotationMetadata.EMPTY_METADATA, Collections.emptyMap(), Argument.ZERO_ARGUMENTS);
     }
 
+    /**
+     * Creates a new argument for the type of the given instance.
+     *
+     * @param instance The argument instance
+     * @param <T>      The generic type
+     * @return The argument instance
+     * @since 4.6
+     */
+    @NonNull
+    static <T> Argument<T> instanceType(T instance) {
```

Comment:
add a null check to the instance argument and nullability annotation. Rename to `forInstance`